<launch>

  <!-- important generally, but specifically utilised by the current app manager -->
  <param name="robot/name" value="$(optenv ROBOT turtlebot)"/>
  <param name="robot/type" value="turtlebot"/>

  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher">
  </node>

  <param name="robot_description" command="cat $(find igibson-ros)/turtlebot/turtlebot.urdf"/>

  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher">
  </node>

  <node name="turtlebot_gibson_sim" pkg="igibson-ros" type="turtlebot_rgbd.py" output="screen">
  </node>

  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find igibson-ros)/rviz/robot.rviz" />

  <!-- Velocity muxer -->
  <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="cmd_vel_mux"
        args="load yocs_cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager">
    <param name="yaml_cfg_file" value="$(find turtlebot_bringup)/param/mux.yaml" />
    <remap from="cmd_vel_mux/output" to="mobile_base/commands/velocity"/>
  </node>

 <include file="$(find turtlebot_teleop)/launch/keyboard_teleop.launch">
 </include>

 <node pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" name="depthimage_to_laserscan"  output="screen">
    <param name="scan_height" value="2"/>
    <param name="output_frame_id" value="/camera_depth_frame"/>
    <param name="range_min" value="0.45"/>
    <remap from="image" to="/gibson_ros/camera/depth/image"/>
    <remap from="scan" to="/gibson_ros/laserscan"/>
  </node>

  <node pkg="nodelet" type="nodelet" args="manager"
        name="record_player_manager" output="screen"/>
        
  <node pkg="nodelet" type="nodelet" name="cloudify"
        args="load depth_image_proc/point_cloud_xyzrgb record_player_manager --no-bond" output="screen">
    <remap from="depth_registered/image_rect" to="/gibson_ros/camera/depth/image"/>
    <remap from="depth_registered/points" to="/gibson_ros/camera/depth_registered/points"/>
    <remap from="rgb/image_rect_color" to="/gibson_ros/camera/rgb/image"/>
    <remap from="rgb/camera_info" to="/gibson_ros/camera/depth/camera_info"/>
  </node>

</launch>